pkg/messaging/rabbitmq/publisher.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package rabbitmq
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+var _ messaging.Publisher = (*publisher)(nil)
+
+type publisher struct {
+	conn *amqp.Connection
+	ch   *amqp.Channel
+}
+
+// Publisher wraps messaging Publisher exposing
+// Close() method for RabbitMQ connection.
+type Publisher interface {
+	messaging.Publisher
+	Close()
+}
+
+// NewPublisher returns RabbitMQ message Publisher.
+func NewPublisher(url string) (Publisher, error) {
+	endpoint := fmt.Sprintf("amqp://%s", url)
+	conn, err := amqp.Dial(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	ch, err := conn.Channel()
+	if err != nil {
+		return nil, err
+	}
+	ret := &publisher{
+		conn: conn,
+		ch:   ch,
+	}
+
+	return ret, nil
+}
+
+func (pub *publisher) Publish(topic string, msg messaging.Message) error {
+	if topic == "" {
+		return errEmptyTopic
+	}
+	data, err := proto.Marshal(&msg)
+	if err != nil {
+		return err
+	}
+	subject := fmt.Sprintf("%s.%s.%s", exchange, chansPrefix, topic)
+	if err := pub.ch.ExchangeDeclare(subject, exchangeKind, true, false, false, false, nil); err != nil {
+		return err
+	}
+
+	err = pub.ch.Publish(
+		subject,
+		routingKey,
+		mandatory,
+		immediate,
+		amqp.Publishing{
+			Headers:     amqp.Table{},
+			ContentType: "text/plain",
```

Comment:
Is this type correct? Since our message is binary, it looks like `application/octet-stream` is more suitable for us.

Reply:
Yes `application/octet-stream` is more suitable